src/lib/dialog/dialog-config.ts

```diff
@@ -3,17 +3,25 @@ import {ViewContainerRef} from '@angular/core';
 /** Valid ARIA roles for a dialog element. */
 export type DialogRole = 'dialog' | 'alertdialog'
 
+/** Interface for a dialog configuration object. */
+export interface MdDialogConfigObject {
+  viewContainerRef: ViewContainerRef;
 
+  /** The ARIA role of the dialog element. */
+  role?: DialogRole;
+}
 
 /**
  * Configuration for opening a modal dialog with the MdDialog service.
  */
-export class MdDialogConfig {
+export class MdDialogConfig implements MdDialogConfigObject {
   viewContainerRef: ViewContainerRef;
-
-  /** The ARIA role of the dialog element. */
   role: DialogRole = 'dialog';
 
+  constructor(config?: MdDialogConfigObject) {
+    Object.assign(this, config);
+  }
```

Comment:
I see what you mean. I want to keep it such the preferred API is 
```ts
dialog.open(SomeComponent, {
  someConfig: true
});
```
And all the defaults are applied. 

How about just having a method in `MdDialog` like `_applyConfigDefaults` that takes the config and applies the default values for things that are unset?